apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -763,6 +763,35 @@ namespace MWMechanics
 
         int school = 0;
 
+        if(mCaster.getClass().isActor()
+                && MWBase::Environment::get().getWorld()->getStore().get<ESM::MagicEffect>()
+                .find(ESM::MagicEffect::WaterWalking))
+        {
+            const MWWorld::CellStore *cell = mCaster.getCell();
+            
+            if(cell->getCell()->hasWater())
+            {
+                const ESM::Position casterPosition = mCaster.getRefData().getPosition();
+                const float casterHeight = mCaster.getCellRef().getScale();
```

Comment:
getScale() isn't the actor's height, it's a scaling factor applied to the model (which you'll find is usually 1).

To get the actor height, you would need to get it from the PhysicsSystem::getHalfExtents, then multiply by getScale().